src/api/events.ts

```diff
@@ -2,32 +2,24 @@ import firebase from "firebase/app";
 
 import { VenueEvent } from "types/venues";
 
-import { withId, WithId, withVenueId, WithVenueId } from "utils/id";
-import { asArray } from "utils/types";
+import { withId, WithId, WithVenueId } from "utils/id";
 
-import { getVenueRef } from "./venue";
-
-export const getVenueEventCollectionRef = (venueId: string) =>
-  getVenueRef(venueId).collection("events");
-
-export const fetchVenueEvents = async (
-  venueId: string
-): Promise<WithVenueId<WithId<VenueEvent>>[]> =>
-  getVenueEventCollectionRef(venueId)
+export const fetchAllVenueEvents = async (
+  sovereignVenueId: string
+): Promise<WithVenueId<WithId<VenueEvent>>[]> => {
+  const eventsSnapshot = await firebase
+    .firestore()
+    .collectionGroup("events")
+    .where("sovereignVenueId", "==", sovereignVenueId)
     .withConverter(venueEventWithIdConverter)
-    .get()
-    .then((docSnapshot) =>
-      docSnapshot.docs.map((venueEvent) =>
-        withVenueId(venueEvent.data(), venueId)
-      )
-    );
+    .get();
 
-export const fetchAllVenueEvents = async (
-  venueIdOrIds: string | string[]
-): Promise<WithVenueId<WithId<VenueEvent>>[]> =>
-  Promise.all(asArray(venueIdOrIds).map(fetchVenueEvents)).then((result) =>
-    result.flat()
-  );
+  return eventsSnapshot.docs
+    .map((venueEvent) => venueEvent.data())
+    .filter((venueEvent) => venueEvent.venueId) as WithVenueId<
+    WithId<VenueEvent>
+  >[];
+};
```

Comment:
Ignoring the `fetchAllVenueEvents` part which is fine to remove. Why was the `.then` removed + separated like this? It should have just been left inline as it was by the looks of things.